### PR TITLE
#912: Fixed image located in the media gallery

### PR DESCRIPTION
**Pull request by mmezhensky**
### Description (*)

This PR fixes the image located in the media gallery.

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#912: Locate Adobe Stock grid functionality has to be adjusted to work with both media galleries

### Manual testing scenarios (*)
1. Login to Admin panel
2. Open Media Gallery (i.e. Catalog -> Category -> expand Content -> Select from Gallery button)
3. Click "Search Adobe Stock"
4. Open preview for an image that was saved from the Adobe Stock previously
5. Click `Open in Media Gallery` button (this image should be selected in the media gallery)

### CODE REVIEW

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @mmezhensky thanks for the pull request! There is an issue with locating the image in the new media gallery if another folder was selected before locating (the folder not containing an image).

Looks like the https://github.com/magento/adobe-stock-integration/blob/db7e9a6626b77e36ef6ff0dab3dd2bfb3002f652/AdobeStockImageAdminUi/view/adminhtml/web/js/media-gallery.js#L64 line is execute before the media gallery view is updated when the folder selection is changed.

This results in `false` returned from the `mediaGallery.locate` call triggering the `mediaGallery.notLocated()` call

Can you please take a look at this issue?

**Comment by mmezhensky** [comment]
> Hi @mmezhensky thanks for the pull request! There is an issue with locating the image in the new media gallery if another folder was selected before locating (the folder not containing an image).
> 
> Looks like the
> 
> https://github.com/magento/adobe-stock-integration/blob/db7e9a6626b77e36ef6ff0dab3dd2bfb3002f652/AdobeStockImageAdminUi/view/adminhtml/web/js/media-gallery.js#L64
> 
> line is execute before the media gallery view is updated when the folder selection is changed.
> This results in `false` returned from the `mediaGallery.locate` call triggering the `mediaGallery.notLocated()` call
> 
> Can you please take a look at this issue?

@sivaschenko 
fixed, please check. Tnx!

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/preview/actions.js [hunk 1]
```diff
@@ -99,11 +119,71 @@ define([
...
+                        return;
+                    }
+
+                    self.selectRecord(image);
+                    subscription.dispose();
+                });
+            }
+
+            if (imageFolders.length) {
+                imageFolders[0].click();
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/preview/actions.js]: If the directory with the image was selected before the locate action was called - the selection of the directory is dropped after this click.

Is it possible to use `MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js.setActiveNodeFilter` method call to select the required directory by path? (even without searching for the folder beforehand)

File: AdobeStockImageAdminUi/view/adminhtml/web/js/media-gallery.js [hunk 2]
```diff
@@ -56,19 +76,11 @@ define([
...
-                return locatedImage.length ?  locatedImage : false;
-            }
-
-            $.ajaxSetup({
-                async: true
+            imageFolder = $('.jstree a').filter(function () {
+                return $.trim($(this).text()) === imageFolderName;
             });
+
+            return imageFolder;
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/media-gallery.js:85]: I'd move the following code snippet to this method and renamed it to `selectFolder`

            if (imageFolder.length) {
                imageFolder[0].click();

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the updates @mmezhensky ! Please see my review comments

Also can you please run the eslint static tests on changed js files.

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/preview/actions.js (hunk 1 above)
Comment (mmezhensky) [reply AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/preview/actions.js]: Fixed and I added small change in the `setActiveNodeFilter` function, please check. 
Thank you!

File: AdobeStockImageAdminUi/view/adminhtml/web/js/media-gallery.js (hunk 2 above)
Comment (mmezhensky) [reply AdobeStockImageAdminUi/view/adminhtml/web/js/media-gallery.js:85]: done, please check. Tnx!

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**